### PR TITLE
Fix wrong tooltip location if tooltipped text has line breaks

### PR DESCRIPTION
**Pull request by yamgent**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Bug fix

Fixes MarkBind/markbind#338.

**What is the rationale for this request?**
Tooltip location is wrong when the tooltipped text wraps to multiple lines.

**What changes did you make? (Give an overview)**
Changed the logic such that the position is always determined by the top-left of the tooltippped-text, and remove the involvement of the width and height as much as possible.

**Provide some example code that this change will affect:**
NA

**Is there anything you'd like reviewers to focus on?**
1. Whether the tooltip looks better with the new implementation.

**Testing instructions:**
1. Run `npm run build`, copy `vue-strap.min.js` over to MarkBind.
2. Run `markbind serve docs` and ensure that nothing goes wrong.
3. Alternatively, try with this content `index.md`:

```html
<tooltip content="hi"><i>coupling</i></tooltip>
<tooltip placement="bottom" content="hi"><i>coupling</i></tooltip>
<tooltip placement="left" content="hi"><i>coupling</i></tooltip>
<tooltip placement="right" content="hi"><i>coupling</i></tooltip>

cupidatat non proident, sunt in culpa qui <tooltip content="hi"><i> quis nostrud exercitation ullamco laboris nisi cillum dolore eu fugiat nulla pariatur. Excepteur sint</i></tooltip> cillum dolore eu fugiat nulla pariatur. dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

cupidatat non proident, sunt in culpa qui <tooltip placement="bottom" content="hi"><i> quis nostrud exercitation ullamco laboris nisi cillum dolore eu fugiat nulla pariatur. Excepteur sint</i></tooltip> cillum dolore eu fugiat nulla pariatur. dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

cupidatat non proident, sunt in culpa qui <tooltip placement="left" content="hi"><i> quis nostrud exercitation ullamco laboris nisi cillum dolore eu fugiat nulla pariatur. Excepteur sint</i></tooltip> cillum dolore eu fugiat nulla pariatur. dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

cupidatat non proident, sunt in culpa qui <tooltip placement="right" content="hi"><i> quis nostrud exercitation ullamco laboris nisi cillum dolore eu fugiat nulla pariatur. Excepteur sint</i></tooltip> cillum dolore eu fugiat nulla pariatur. dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

<tooltip placement="left" content="bye">This tooltip is on the left.</tooltip>

<div align="right">
  <tooltip placement="right" content="bye">This tooltip is on the right.</tooltip>
</div>

More about <trigger for="tt:trigger_id">trigger</trigger>.
<tooltip id="tt:trigger_id" content="This tooltip triggered by a trigger"></tooltip>
<br>
This is the same <trigger for="tt:trigger_id">trigger</trigger> as last one.

Week         | Activities
-------------|-----------
6 (mid-v1.1) |  <tooltip content="User Guide">User Guide</tooltip>

Week         | Activities
-------------|-----------
6 (mid-v1.1) |  <popover content="User Guide">User Guide</popover>
```

### CODE REVIEW

**Comment by yamgent** [comment]
The bootstrap-vue component does not cater to this use-case:

```html
More about <trigger for="tt:trigger_id">trigger</trigger>.
<tooltip id="tt:trigger_id" content="This tooltip triggered by a trigger"></tooltip>
This is the same <trigger for="tt:trigger_id">trigger</trigger> as last one.
```

This use case was possible previously, because vue-strap is able to manipulate the position of the tooltip directly, so sharing tooltip is possible.

In bootstrap-vue, the tooltip is always tied to one particular tooltipped text.

File: src/utils/popoverMixins.js [hunk 1]
```diff
@@ -100,34 +100,55 @@ export default {
...
+      if (popoverViewportOffset.left < 0) {
+        this.position.left -= popoverViewportOffset.left;
+      }
+      if (popoverViewportOffset.top < 5) {
+        this.position.top -= popoverViewportOffset.top - 5;
+      }
+      if (popoverViewportOffset.left > window.innerWidth - popover.offsetWidth) {
+        this.position.left -= popoverViewportOffset.left - (window.innerWidth - popover.offsetWidth);
+      }
+      // TODO This offset does not work, it will just go off the screen
```
Comment (yamgent) [inline src/utils/popoverMixins.js]: @nusjzx As mentioned in person, this is the part of the code that does not work.

No rush into looking at this, you can focus on your other work first (this use case is not in `website-base`, so not a very big issue atm).

Thanks for taking a look at this!

Comment (nusjzx) [reply src/utils/popoverMixins.js]: This part of code should be put after line 90, where all the adjustment are done. That will solve the problem.

Comment (yamgent) [reply src/utils/popoverMixins.js]: @nusjzx thanks for checking! The fix works now, will tidy up the PR.

**Comment by yamgent** [comment]
**Update:**
* Fixed position bugs.
* Popover should now always be inside the viewport.

**Comment by yamgent** [comment]
The problem with MarkBind/markbind#338 (the tooltip misplacement) is fixed with the current iteration of the code. However, the tooltips in the documentations are messed up now.

No idea how to proceed with the fix.

Attached GIF of the problem:

tooltip_bug

**Comment by yamgent** [comment]
As discussed, we will just stick with placing the tooltip under the mouse pointer.

Will be closing this PR so that the changes here are archived, in case there's interest in revisiting this issue in the future.

**Pull request closed without merging**